script_editor/app/views/plays/show.html.erb

```diff
@@ -319,159 +339,164 @@
+                                        <%  # add a special class marker to this direction showing it is inside a speech.
+                                            #this enables the direction to be cut when a whole speech is cut by clicking on the 
+                                            #character name %>
+                                        <button class="stage inside" data-cut="false" data-display="true">
+                                            <%= stage.inner_text %>
                                         </button>
-                                    <% else %>
-                                        <% if word.inner_text != "" && word.inner_text != " " %>
-                                            <button class="word" data-cut="false" data-display="true" data-lineNum=<%= lineNum %>>
-                                                <%= word.inner_text %>
-                                            </button>
-                                        <% end %>
+                                        <br>
+
+                                        <% stages.delete(stage) %>
                                     <% end %>
                                 <% end %>
-                            <% end %>
-                        <% end %>
 
-                        <% # BREAK FOR NEW LINE %>
-                        <br class = "newLine" data-lineNum=<%= lineNum %>  >
+                                <% # get line number %>
+                                <% lineNum = ms.attr("n").to_s.split(".")[2] %>
+                                <% # get corresps per milestone in array %>
+                                <% wordIDs = ms.attr("corresp").to_s.split(" ") %>
+                                <% # remove #s %>
+                                <% wordIDs = wordIDs.map { |w| w.gsub("#","")} %>
+                                <% # Search spwords for corresponding w/c/pc %>
+                                
+                                <% # DISPLAY LINE NUMBER %>
+                                <% if lineNum.to_i % 5 == 0 %>
+                                    <p class="lineNum" id = "originalLineNum">
+                                        <%= lineNum %>
+                                    </p>
+                                <% end %>
 
+                                <% wordIDs.each do |id| %>
+                                    <% spwords.each do |word| %>
+                                        <% if word.attr('xml:id').to_s == id %>
+                                            <% # DISPLAY EACH WORD %>
+                                            <% if (word.inner_text == ".") || (word.inner_text == ",") || (word.inner_text == "?") || (word.inner_text == "!") || (word.inner_text == ";") || (word.inner_text == ":") %>
+                                                <button class="punc" data-cut="false" data-display="true" data-lineNum=<%= lineNum %>>
```

Comment:
I know this isn't code you wrote, but do you think using the .include() method that you suggested to Montana would be useful here? Or is that significantly more difficult in HTML than JS?